packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -686,6 +688,36 @@ class ScaffoldState extends State<Scaffold> {
     );
   }
 
+  // IOS-specific back gesture.
+
+  final GlobalKey _backGestureKey = new GlobalKey();
+  NavigationGestureController _backGestureController;
+
+  bool _shouldHandleBackGesture() {
+    return Navigator.canPop(context);
+    // return Theme.of(context).platform == TargetPlatform.iOS && Navigator.canPop(context);
+  }
+
+  void _handleDragStart(DragStartDetails details) {
+    _backGestureController = Navigator.of(context).startPopGesture();
+  }
+
+  void _handleDragUpdate(DragUpdateDetails details) {
+    _backGestureController?.dragUpdate(
+      details.primaryDelta / MediaQuery.of(context).size.width
```

Comment:
i think this should use `findRenderObject().size.width` rather than the screen width.